dg.go

```diff
@@ -24,20 +25,32 @@ type directedGraph[NodeType any] struct {
 }
 
 func (d *directedGraph[NodeType]) Mermaid() string {
-    result := []string{}
-    result = append(result, "flowchart TD")
-    result = append(result, "subgraph input")
-    for source := range d.connectionsFromNode {
-	if strings.HasPrefix(source, "input") {
-            result = append(result, source)
-	}
+    result := []string{
+        "%% Mermaid markdown workflow",
+        "flowchart LR",
+        "%% Success path",
     }
-    result = append(result, "end")
+
+    errorPath := []string{"%% Error path"}
+    errorPathRegex, _ := regexp.Compile("error|crashed|failed|deploy_failed")
```

Comment:
It may make sense to make this a global, so it just runs once instead of every time this is called.

Reply:
Another concern is this is hard-coded. I wonder if it would be worth the effort to make this an input. Like a list of strings that matches an error type, or even more generically, a map of category type to a pattern or list of values that matches that category.

What do you think, @webbnh?

Reply:
I thought about the hard-coded problem, but I don't know enough of the code base or golang to see a better approach right now.

Reply:
> It may make sense to make this a global, so it just runs once instead of every time this is called.

Done.

Reply:
> Another concern is this is hard-coded. I wonder if it would be worth the effort to make this an input. Like a list of strings that matches an error type, or even more generically, a map of category type to a pattern or list of values that matches that category.

Yeah, I wondered about the hardcoded list, too.

How many difference patterns are we likely to have, and what informs the change?

Reply:
> Yeah, I wondered about the hardcoded list, too.
> 
> How many difference patterns are we likely to have, and what informs the change?

These are workflow step "nodes" and could change with any alterations to the workflow. AFAIK we don't define anywhere else which nodes are the "error" path and which are the "success" path like this. In principle, it would make more sense for those paths to be defined somehow in the workflow and inherited here, but for the sake of this improvement alone, I'm inclined to keep it this way for now and avoid external change dependencies just for the sake of some improved debug logging. But maybe I should open an issue for the engine to define these paths there in the future.

Reply:
> I wonder if it would be worth the effort to make this an input.

Unless this is currently or soon to be called from multiple places, moving the list from a global to the caller is just kicking the can up the call stack -- it doesn't really solve any problem (and introduces additional complexity).

The answer might be different if there were a canonical list defined somewhere else, but I'm not hearing that there is.  And, as Dustin implies, we can go with the hard-coded list in this PR and iterate once we have a better solution.